libraries/rush-lib/src/logic/operations/OperationMetadataManager.ts

```diff
@@ -112,31 +141,46 @@ export class OperationMetadataManager {
 
   public async tryRestoreAsync({
     terminal,
-    logPath,
-    errorLogPath
+    terminalProvider,
+    errorLogPath,
+    logChunksPath
   }: {
+    terminalProvider: ITerminalProvider;
     terminal: ITerminal;
     logPath: string;
     errorLogPath: string;
+    logChunksPath: string;
   }): Promise<void> {
     await this.stateFile.tryRestoreAsync();
 
-    // Append cached log into current log file
-    terminal.writeLine(`Restoring cached log file at ${this._logPath}`);
     let logReadStream: fs.ReadStream | undefined;
     try {
-      logReadStream = fs.createReadStream(this._logPath, {
-        encoding: 'utf-8'
-      });
-      for await (const data of logReadStream) {
-        terminal.write(data);
+      if (await FileSystem.existsAsync(logChunksPath)) {
+        const logChunks: ILogChunkStorage = (await JsonFile.loadAndValidateAsync(
+          logChunksPath,
+          OperationMetadataManager._logChunkJsonSchema
+        )) as ILogChunkStorage;
+        for (const chunk of logChunks.chunks) {
+          const { text, kind } = chunk;
+          if (kind === TerminalChunkKind.Stderr) {
+            terminalProvider.write(text, TerminalProviderSeverity.error);
```

Comment:
What about warnings?

Reply:
This is pulled from ShellOperationPlugin which only logs with severity log and error for stdout and stderr respectively.